### PR TITLE
Use the same version of `marked` in our build and our plugins

### PR DESCRIPTION
**Pull request by colinrotherham**
Unblocks `marked@8.0.0` via https://github.com/alphagov/govuk-design-system/pull/3123 and closes:

* https://github.com/alphagov/govuk-design-system/issues/2767

HTML changes include:

1. Automatic "smart" punctuation is unescaped `&#8216;` → `‘` everywhere
2. Automatic "smart" punctuation is now applied to Nunjucks rendered HTML
3. Indentation is better preserved around code blocks
4. Macro options Markdown now shares the same renderer

### CODE REVIEW

**Comment by colinrotherham** [comment]
Need to find a suitable replacement for `sanitize: true` to fix `&` → `&amp;`

* [DOMPurify](https://github.com/cure53/DOMPurify) – Aggressively removes "unsafe" HTML
* [sanitize-html](https://github.com/apostrophecms/sanitize-html) – Converts void elements to self-closing

**Comment by colinrotherham** [comment]
We don't use full HTML sanitisation currently at the suggestions made quite extreme changes

I've instead used `walkTokens` in https://github.com/alphagov/govuk-design-system/pull/3130/commits/4a1d511bd45a8ffaf92b1e4da1e52a1dba7b25b7 to replace those stray `&` → `&amp;`

Ready to review

**Comment by domoscargin** [comment]
This is a pretty massive diff to scan through, though it looks like two main things:

1. The formatting has changed in some places (ie: looks nicer), eg:

```diff
diff --color -r .cache/public/styles/spacing/index.html deploy/public/styles/spacing/index.html
1037,1040c1033,1036
< <span class="hljs-tag">&lt;<span class="hljs-name">div</span> <span class="hljs-attr">class</span>=<span class="hljs-string">&quot;govuk-!-padding-bottom-1&quot;</span>&gt;</span><span class="hljs-tag">&lt;/<span class="hljs-name">div</span>&gt;</span>
< </code></pre>
<   </div>
< </div></div>
---
> <span class="hljs-tag">&lt;<span class="hljs-name">div</span> <span class="hljs-attr">class</span>=<span class="hljs-string">&quot;govuk-!-padding-bottom-1&quot;</span>&gt;</span><span class="hljs-tag">&lt;/<span class="hljs-name">div</span>&gt;</span></code></pre>
>       </div>
>     </div>
> </div>
```

2. We've replaced encoded characters with HTML entities:

```diff

diff --color -r .cache/public/accessibility/index.html deploy/public/accessibility/index.html
968,969c968,969
< <li>Simple and intuitive use – The design is easy to understand, regardless of the user’s experience, knowledge, language skills, or current concentration level.</li>
< <li>Perceptible information – The design communicates necessary information effectively to the user, regardless of ambient conditions or the user’s sensory abilities.</li>
---
> <li>Simple and intuitive use – The design is easy to understand, regardless of the user&#8217;s experience, knowledge, language skills, or current concentration level.</li>
> <li>Perceptible information – The design communicates necessary information effectively to the user, regardless of ambient conditions or the user&#8217;s sensory abilities.</li>
```

On (1), the formatting looks nicer on the whole, and doesn't seem to have broken anything at a glance, so probably fine.

(2) is a bit more iffy as to whether it's desirable. Is there a way to make `markedSmartypants` behave like [`smartypantsu` ](https://github.com/othree/smartypants.js#es6-module)(ie: compile to UTF)?

**Comment by colinrotherham** [comment]
Thanks @domoscargin

Yeah we started seeing HTML entities from [marked-smartypants](https://github.com/markedjs/marked-smartypants) in https://github.com/alphagov/govuk-design-system/pull/3002 but not everywhere

With this PR any Markdown that used `{ smartypants: true }` now uses entities too

I'd prefer the same but some digging shows it's the default:

* https://github.com/markedjs/marked/issues/561#issuecomment-371576680

The recommended [sanitize-html](https://github.com/apostrophecms/sanitize-html) unescapes all entities but then you get self-closing tags `<hr />` back

**Comment by colinrotherham** [comment]
@domoscargin Or could [overwrite `hooks`](https://github.com/markedjs/marked-smartypants/blob/9862ad249bc4309795c7fbf0115bfaa60ffe2e6e/src/index.js#L25-L29) and call [`smartypantsu`](https://github.com/othree/smartypants.js#es6-module) instead?

File: lib/marked/extension.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+exports.headingIds = gfmHeadingId()
+
+// Extension: Code syntax highlighter
+exports.highlight = markedHighlight({ highlight })
+
+// Extension: Code syntax highlighter
+exports.smartyPants = {
+  ...markedSmartypants(),
+
+  // Use UTF-8 characters not HTML entities
```
Comment (domoscargin) [inline lib/marked/extension.js:19]: :+1:

**Comment by domoscargin** [comment]
Diff appears to mostly be:

- formatting changes to the code blocks,
- nice pick ups from smartypants that we've missed
  -  We should edit [this example code](https://github.com/alphagov/govuk-design-system/blob/f5a4007208f78fd5f0189bfbb48e562407724001/views/partials/_example.njk#L87-L93), since there's a LOT of instances of it that smartpants has to transform. There's a few others in the diff, but none of them appear 312 times 😆  Quite a few of them come from the `govuk-frontend` YAML properties, so we could potentially `grep` and replace there to reduce transformations further.

However, there is some weirdness, like:

## Replacing `--` with `–` in some examples
```diff
diff --color -r .cache/public/get-started/updating-your-code/index.html deploy/public/get-started/updating-your-code/index.html
1146c1153
<       <td class="govuk-table__cell ">govuk-section-break<br>govuk-section-break--visible<br>govuk-section-break--xl<br>govuk-section-break--l<br>govuk-section-break--m</td>
---
>       <td class="govuk-table__cell ">govuk-section-break<br>govuk-section-break–visible<br>govuk-section-break–xl<br>govuk-section-break–l<br>govuk-section-break–m</td>
1167c1174
<       <td class="govuk-table__cell ">govuk-list<br>govuk-list--bullet</td>
---
>       <td class="govuk-table__cell ">govuk-list<br>govuk-list–bullet</td>
1171c1178
<       <td class="govuk-table__cell ">govuk-list<br>govuk-list--number</td>
---
>       <td class="govuk-table__cell ">govuk-list<br>govuk-list–number</td>
1242c1249
<       <td class="govuk-table__cell ">govuk-label--s</td>
---
>       <td class="govuk-table__cell ">govuk-label–s</td>
```
Suspect we just want to wrap these in `<code>`

## Some component options tables are weird?
For some reason, the Notification Banner and Summary List component Nunjucks options tables are showing up as big diffs (though they look like they're just getting a bit of formatting). The Summary List options tables are about 5000+ lines of the diff file.

Is there something funky with the page code that's affecting these tables versus other components where the tables are not getting reformatted?

**Comment by domoscargin** [comment]
Ha, Dependabot just burns everything on the way out? Nice.

**Comment by colinrotherham** [comment]
How annoying, can't re-open?

**Pull request closed without merging**